File: CeliTax/TaxYearBuilder.m
//
//  TaxYearBuilder.m
//  CeliTax
//
//  Created by Leon Chen on 2015-07-20.
//  Copyright (c) 2015 CraveNSave. All rights reserved.
//

#import "TaxYearBuilder.h"

@implementation TaxYearBuilder

- (TaxYear *) buildTaxYearFrom: (NSNumber *) taxYearNumber
{
    if (![taxYearNumber isKindOfClass: [NSNumber class]])
    {
        return nil;
    }
    
    TaxYear *taxYear = [[TaxYear alloc] init];
    
    taxYear.taxYear = taxYearNumber.integerValue;
    
    return taxYear;
}

@end

File: CeliTax/TaxYearBuilder.swift
//
//  TaxYearBuilder.swift
//  CeliTax
//
//  Created by Leon Chen on 2015-11-01.
//  Copyright © 2015 CraveNSave. All rights reserved.
//

import Foundation

@objc
class TaxYearBuilder : NSObject //TODO: Remove Subclass to NSObject when the entire app has been converted to Swift
{
    func buildTaxYearFrom(taxYearNumber : Int) -> TaxYear
    {
        let taxYear : TaxYear = TaxYear()
        
        taxYear.taxYear = taxYearNumber
        
        return taxYear
    }
}